### ISSUE TITLE
Unused variables only appear in HTML report

### ISSUE THREAD

**Issue by happygiraffe** [douglascrockford/JSLint#112]
Presently, unused variables are only reported as part of the `JSLINT.report()` call.  e.g.

```
function foo(a, b) {
    return a;
}
```

Produces:
- Unused variable: b 1 foo

However, variables which are undefined at the point of use merit an error, as well as being called out in report().  e.g.

```
function foo() {
    return a;
}
```

Produces:
- Problem at line 2 character 12: 'a' was used before it was defined.
- Undefined variable: a 1 foo

I think that either unused variables should be reported as an error, or undefined variables should be moved out of the report().

Thanks,
-Dom
